test/mlir/onnx/onnx_lowering_with_canonicalize.mlir

```diff
@@ -2329,9 +2329,69 @@ func.func @pad_constant_mode(%arg0: tensor<1x3x4x5xf32>, %arg1: tensor<8xi64>, %
 // CHECK:           return [[RES_]] : memref<?x?x?x?xf32>
 // CHECK:         }
 }
+// -----
+
+func.func @pad_constant_mode_const_arg_format(%arg0: tensor<1x3x4x5xf32>, %arg1: tensor<8xi64>, %arg2: tensor<1xf32>) -> tensor<*xf32> {
+  %0 = "onnx.Pad"(%arg0, %arg1, %arg2) {mode = "constant"} : (tensor<1x3x4x5xf32>, tensor<8xi64>, tensor<1xf32>) -> tensor<*xf32>
```

Comment:
I don't really know how to check. Neither the documentation nor the code is helping since the documentation allows every type of tensor as input, which I already have some doubt on the validity. The code for the op or its lowering doesn't constrain on that either.